src/PrestaShopBundle/Translation/Provider/ExternalModuleLegacySystemProvider.php

```diff
@@ -0,0 +1,244 @@
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultCatalogue($empty = true)
+    {
+        $defaultCatalogue = $this->moduleProvider
+            ->setLocale($this->locale)
+            ->getDefaultCatalogue()
+        ;
+
+        try {
+            $additionalDefaultCatalogue = $this->legacyModuleExtractor->extract($this->moduleName, $this->getLocale());
+            $defaultCatalogue->addCatalogue($additionalDefaultCatalogue);
+        } catch (UnsupportedLocaleException $exception) {
+            // Do nothing as support of legacy file is deprecated
+        }
+
+        if ($empty) {
+            $defaultCatalogue = $this->emptyCatalogue($defaultCatalogue);
+        }
+
+        return $defaultCatalogue;
+    }
+
+    public function getXliffCatalogue()
+    {
+        return new MessageCatalogue($this->locale);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultResourceDirectory()
+    {
+        return $this->resourceDirectory . DIRECTORY_SEPARATOR . $this->moduleName . DIRECTORY_SEPARATOR . 'translations' . DIRECTORY_SEPARATOR;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getResourceDirectory()
+    {
+        return $this->getDefaultResourceDirectory();
+    }
+
+    /**
+     * @return string returns the module domain
+     */
+    private function getModuleDomain()
+    {
+        return 'Modules' . Container::camelize($this->moduleName);
+    }
+
+    /**
+     * The extracted catalogue come from translations folder.
+     * The default catalogue come from module code parsing.
+     *
+     * @return MessageCatalogue
+     *
+     * @throws \Exception
+     */
+    public function getLegacyCatalogue()
+    {
+        $catalogueFromFiles = $this->moduleProvider->getXliffCatalogue();
+
+        try {
+            $defaultCatalogue = $this->getDefaultCatalogue();
+
+            $extractedCatalogue = $this->legacyFileLoader->load(
+                $this->getDefaultResourceDirectory(),
+                $this->locale,
+                $this->getModuleDomain()
+            );
+
+            $legacyFilesCatalogue = new MessageCatalogue($this->locale);
+
+            $translations = $defaultCatalogue->all($this->getModuleDomain());
+
+            foreach (array_keys($translations) as $translationKey) {
+                $legacyKey = md5($translationKey);
+
+                if ($extractedCatalogue->has($legacyKey, $this->getModuleDomain())) {
+                    $legacyFilesCatalogue->set(
+                        $translationKey,
+                        $extractedCatalogue->get($legacyKey, $this->getModuleDomain()),
+                        $this->getModuleDomain()
+                    );
+                }
+            }
+
+            $catalogueFromFiles->addCatalogue($legacyFilesCatalogue);
+        } catch (UnsupportedLocaleException $exception) {
+            // Do nothing as support of legacy file is deprecated
```

Comment:
Some modules don't have legacy files: it's not common in the Core, but it is the way to go for the community modules.

As the support of this legacy format is deprecated, I don't see the value of displaying an error message to the developer/merchant here...